src/js/utils/fileCommon.js

```diff
@@ -0,0 +1,132 @@
+    const hhmmss = `${zeroPad(date.getHours(), 2)}${zeroPad(date.getMinutes(), 2)}${zeroPad(date.getSeconds(), 2)}`;
+    filename = `${filename}_${yyyymmdd}_${hhmmss}`;
+
+    return `${filename}.${suffix}`;
+}
+
+const getDefaultPath = () => {
+    let location;
+    ConfigStorage.get([GUI.active_tab], (result) => location = (result) ? result[GUI.active_tab] : '');
+    return location;
+};
+
+const setDefaultPath = (location) => ConfigStorage.set({[GUI.active_tab]: location});
+
+/*
+* NWjs has some extra attributes extending input element:
+*
+* nwdirectory => returns path of a directory (files picking disabled - not much of use without implementing our own file browser).
+* nwsaveas => opens 'save as' dialog with suggested filename
+* accepts => when loading files limits the used extensions and data type - example ".txt, .config, plain/text".
+* nwworkingdir => sets location (no need to set this as it restores the last used value for the tab).
+*/
+const fileDialog = (options, callback) => {
+
+    // discard previous input element
+    const id = document.querySelector('#fileDialog');
+    if (id) {
+        id.parentNode.removeChild(id);
+    }
+    // add new input element
+    const parent = document.querySelector('.fileInput');
+    const input = document.createElement('input');
+    parent.append(input);
+    // add default attributes
+    const newId = document.querySelector('.fileInput input');
+    newId.setAttribute('style', 'display:none;');
+    newId.setAttribute('id', 'fileDialog');
+    newId.setAttribute('type', 'file');
+    newId.setAttribute('nwworkingdir', getDefaultPath());
+    // add custom attributes
+    Object.keys(options).forEach((key) => {
+        newId.setAttribute(key, options[key]);
+    });
+
+    const chooseFile = (name) => {
+        const chooser = document.querySelector(name);
+        chooser.addEventListener("change", (e) => {
+            if (e.target.value) {
+                setDefaultPath(path.dirname(e.target.value));
+                callback(e.target.value);
+                e.target.value = '';
+            }
+        }, false);
+
+        chooser.click();
+    };
+
+    chooseFile('#fileDialog');
+};
+
+export const fileOpen = (accept, callback) => fileDialog({accept: accept}, callback);
+export const fileSave = (filename, callback) => fileDialog({nwsaveas: filename}, callback);
+export const fileRead = (filename, callback) => fs.readFile(filename, 'utf8', (error, data) => callback(error, data));
+export const fileWrite = (filename, content, callback) => fs.writeFile(filename, content, (error) => callback(error));
+
+// TODO: make this function work for fileWriter (which is retired ;-) or find another solution
+export const fileGetEntry = (filename, callback) => fs.root.getFile(filename, {create: true}, (fileEntry) => callback(fileEntry));
+
+const getFullFileName = (location, filename) => `${location}${path.sep}${filename}`;
```

Comment:
We are trying to group all the export sentences at the end of the file, in this way we can see in a look which methods are published by the module.

Reply:
One PR per tab, but already trying on two other tabs because everything depends heavily on `Entry` object it needs another approach using `fs` now we have a `File` object.